src/XrdCl/XrdClAsyncSocketHandler.hh

```diff
@@ -149,67 +149,67 @@ namespace XrdCl
       //------------------------------------------------------------------------
       // Connect returned
       //------------------------------------------------------------------------
-      virtual void OnConnectionReturn();
+      virtual XRootDStatus OnConnectionReturn();
```

Comment:
Andy suggests to add `XRD_WARN_UNUSED_RESULT` to these to ensure we are not ignoring any of the status being returned in other places.